datacube/model/_base.py

```diff
@@ -18,3 +18,6 @@ def ranges_overlap(ra: Range, rb: Range) -> bool:
     if ra.begin <= rb.begin:
         return ra.end > rb.begin
     return rb.end > ra.begin
+
+
+NotType = namedtuple('NotType', 'value')
```

Comment:
I don't like the 'Type' in the name - feels unnecessary and potentially confusing - can we just call it `Not`?